File: AppData/Mods/BUILT-IN/_lua/LuaScript/Uzil/PostProc.lua
local this = {};

function this.inst (key)

	local inst = {};
	inst.key = key;

	--[[ 設置 優先(越小越先) ]]
	inst.setPriority = function (userName, priority)
		this.setPriority(inst.key, userName, priority);
	end

	--[[ 設置效果 ]]
	inst.setEffects = function (userName, effects)
		this.setEffects(inst.key, userName, effects);
	end

	--[[ 移除效果 ]]
	inst.removeEffects = function (userName)
		this.removeEffects(inst.key, userName);
	end

	--[[ 更新 效果 ]]
	inst.updateEffects = function ()
		this.updateEffects(inst.key);
	end

	return inst;
end


--[[ 設置 優先(越小越先) ]]
function this.setPriority (key, userName, priority)
	UZAPI.PostProc.SetPriority(key, userName, priority);
end

--[[ 設置 效果 ]]
function this.setEffects (key, userName, effects)
	UZAPI.PostProc.SetEffects(key, userName, Json.encode(effects));
end

--[[ 移除 效果 ]]
function this.removeEffects (key, userName)
	UZAPI.PostProc.RemoveEffects(key, userName);
end

--[[ 更新 效果 ]]
function this.updateEffects (key)
	UZAPI.PostProc.UpdateEffects(key);
end


return this;

File: Assets/Import_out/Uzil/API/PostProc.cs
using System.Collections.Generic;

using Uzil;
using Uzil.PostProc;

namespace UZAPI {

public class PostProc {

	
	/*======================================Constructor==========================================*/

	/*=====================================Static Members========================================*/

	/*=====================================Static Funciton=======================================*/

	/** 設置 優先 */
	public static void SetPriority (string instID, string userName, float priority) {
		PostProcMgr ppMgr = PostProcMgr.Inst(instID);
		ppMgr.SetPriority(userName, priority);
	}

	/** 設置 效果 */
	public static void SetEffects (string instID, string userName, string effects_str) {
		PostProcMgr ppMgr = PostProcMgr.Inst(instID);

		DictSO effects = DictSO.Json(effects_str);

		foreach (KeyValuePair<string, object> pair in effects) {
			ppMgr.SetEffect(userName, pair.Key, DictSO.Json(pair.Value));
		}
	}

	/** 移除 效果 */
	public static void RemoveEffects (string instID, string userName) {
		PostProcMgr ppMgr = PostProcMgr.Inst(instID);
		ppMgr.RemoveEffects(userName);
	}

	/** 更新 效果 */
	public static void UpdateEffects (string instID) {
		PostProcMgr ppMgr = PostProcMgr.Inst(instID);
		ppMgr.UpdateEffect();
	}

	/*=========================================Members===========================================*/

	/*========================================Components=========================================*/

	/*==========================================Event============================================*/

	/*======================================Unity Function=======================================*/
	
	/*========================================Interface==========================================*/

	/*=====================================Public Function=======================================*/

	/*===================================Protected Function======================================*/
	
	/*====================================Private Function=======================================*/
}


}
